pkg/cloudprovider/suite_test.go

```diff
@@ -197,11 +197,19 @@ var _ = Describe("CloudProvider", func() {
 		Expect(awsEnv.InstanceTypesProvider.UpdateInstanceTypes(ctx)).To(Succeed())
 		Expect(awsEnv.InstanceTypesProvider.UpdateInstanceTypeOfferings(ctx)).To(Succeed())
 	})
-	It("should not proceed with instance creation of nodeClass in not ready", func() {
+	It("should not proceed with instance creation if NodeClass is unknown", func() {
+		nodeClass.StatusConditions().SetUnknown(opstatus.ConditionReady)
+		ExpectApplied(ctx, env.Client, nodePool, nodeClass, nodeClaim)
+		_, err := cloudProvider.Create(ctx, nodeClaim)
+		Expect(err).To(HaveOccurred())
+		Expect(corecloudproivder.IsNodeClassNotReadyError(err)).To(BeFalse())
```

Comment:
can you fix this in the file?
```suggestion
		Expect(corecloudprovider.IsNodeClassNotReadyError(err)).To(BeFalse())
```

Reply:
good eye